Remove google pub sub stuff

brandynbennett opened the pull request:
Start the process of bringing polyn-ruby up to date with the decisions made over the last couple months. 
* Following updated naming conventions
* Focus on NATS as core message system
* Use of a Schema Repository key/value store
* Expecting pre-combined cloud event & json schemas
* etc.

AdamLeMmon01 commented on README.md:
```diff
@@ -153,19 +95,19 @@ would look like:
-  
-  
+
+
```
We must not have rubocop set up on this repo. It would complain about the double blank lines here.

brandynbennett replied:
does rubocop watch markdown files?

AdamLeMmon01 replied:
oh yeah, I guess that's a good point... 😅

AdamLeMmon01 commented on lib/polyn/configuration.rb:
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require_relative "naming"
```
I think we've tried to steer clear of require_relative as a rule. I think in spiff_rb we've opted to just eager load everything up front instead.

brandynbennett replied:
oh interesting, I don't know much about that. Where does the eager loading happen? What problems does `require_relative` cause?

AdamLeMmon01 replied:
I'm less familiar with how Ruby loads it, so someone else will have to help with that. I just know that we got into a situation where it would get caught in a load loop and die...

brandynbennett replied:
oh no! that sounds bad. I'll look into that more

xathien replied:
Rails uses its own Autoloader to locate files. I assumed that being in the `lib` directory maybe sidestepped the loader and that's why all this `require` stuff was put in there, but this comment thread makes me wonder if that assumption was invalid. 😅

AdamLeMmon01 commented on spec/polyn/serializers/json_spec.rb:
```diff
@@ -20,89 +20,85 @@
 # DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
```
The bottom chunk of this license appears to be duplicated for some reason.

AdamLeMmon01 commented on spec/polyn/utils/hash_spec.rb:
```diff
@@ -44,6 +44,22 @@
+      })
+    end
+
+    it "should return if not a hash" do
```
It looked like there was also an array path, but I don't see a test for that path.

brandynbennett replied:
ah, I tested if it could handle an array inside, but didn't do one for a top level array. I'll add that

AdamLeMmon01 reviewed (approved):
left a couple of non-blocker comments. Overall looks good. Good spec coverage!

brandynbennett commented:
@AdamLeMmon01 Thanks for looking through this. I will make some updates

brandynbennett commented on lib/polyn/configuration.rb:
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
```
Good question. My editor keeps adding them in automatically

brandynbennett replied:
Guessing it's a Rubocop rule

brandynbennett commented:
@AdamLeMmon01 I did the following:
* added a spec to test for stringify keys for a top level array
* Removed all the `require_relative` and load them upfront
* Fixed license duplication
